app/services/save_offer_conditions_from_params.rb

```diff
@@ -29,6 +29,9 @@ def conditions
   def serialize_structured_conditions
     return if structured_conditions.blank?
 
+    # Delete all current ske conditions if there is a change of course
+    @offer.ske_conditions.destroy_all if @offer.ske_conditions.any? && structured_conditions.any? { |structured_condition| structured_condition.type == 'SkeCondition' }
```

Comment:
Is this line working as expected? Probably is, but maybe we can extract to a method?

Reply:
:+1: Makes sense to extract a method for this. The second condition seems to be redundant so I've removed that.